src/webrtc/peerconnection.ts

```diff
@@ -104,6 +104,16 @@ export interface PeerConnection<TSignallingMessage> {
 //      3.1. completeConnection_ -> [Fulfill onceConnected]
 export class PeerConnectionClass implements PeerConnection<signals.Message> {
 
+  // Interval, in milliseconds, after which the peerconnection will
+  // terminate if no heartbeat is received from the peer.
+  private static HEARTBEAT_TIMEOUT_MS_ = 10000;
```

Comment:
I think this is very pessimistic.  TCP sockets will survive a 2 minute disconnection.  This means a 10 second network interruption will always kill your proxy session.